### PR TITLE
RavenDB-16782 - support DateTime in the format of "yyyy'-'MM'-'dd'T'HH':'mm':'ss'+'dd':'dd'"

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-16782

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: test/SlowTests/Issues/RavenDB-16782.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+                    result = await session.Query<User>().Where(x => x.DateTime == "2015-10-17T13:28:17-04:00").ToListAsync();
+                    Assert.Equal(0, result.Count);
+                }
+            }
+        }
+
+        private class User
+        {
+            public string DateTime { get; set; }
```
Comment (ryanheath) [inline test/SlowTests/Issues/RavenDB-16782.cs:44]: Maybe I'm missing something here, but shouldn't this be DateTime type?
And if it is supposed to be a string type, how is this test related to DateTime? All I see are string comparisons, no?

Comment (ppekrol) [reply test/SlowTests/Issues/RavenDB-16782.cs:44]: We are skipping here Client's internal behavior where DateTimeOffset is sent using `o` format instead of `ISO-8601` for which we are adding support in this PR.

**Review by arekpalinski — APPROVED** [review]
